Using device: cuda
Found 1 files
Total tokens: 338026
Epoch 1:   0%|                                                                                                                                                                             | 0/9479 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/alexander/Projects/financial_projects/reasonable/train.py", line 213, in <module>
    train_loss = train_one_epoch(model, train_loader, optimizer, scheduler, device, epoch)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/Projects/financial_projects/reasonable/train.py", line 25, in train_one_epoch
    logits = model(x)
             ^^^^^^^^
  File "/home/alexander/miniconda3/envs/pytorch/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/miniconda3/envs/pytorch/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/Projects/financial_projects/reasonable/model.py", line 201, in forward
    x = block(x, mask=mask)
        ^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/miniconda3/envs/pytorch/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/miniconda3/envs/pytorch/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/Projects/financial_projects/reasonable/model.py", line 159, in forward
    x = x + self.attn(self.ln1(x), mask=mask)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/miniconda3/envs/pytorch/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/miniconda3/envs/pytorch/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/Projects/financial_projects/reasonable/model.py", line 103, in forward
    q = apply_rotary_emb(q, self.freqs_cis[:T])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alexander/Projects/financial_projects/reasonable/model.py", line 23, in apply_rotary_emb
    x_rotated = x_complex * freqs_cis
                ~~~~~~~~~~^~~~~~~~~~~
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 48.00 MiB. GPU 0 has a total capacity of 23.76 GiB of which 17.12 MiB is free. Process 477097 has 22.80 GiB memory in use. Including non-PyTorch memory, this process has 948.00 MiB memory in use. Of the allocated memory 489.17 MiB is allocated by PyTorch, and 2.83 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
